pkg/storageincentives/nodestatus.go

```diff
@@ -0,0 +1,162 @@
+// Copyright 2023 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package storageincentives
+
+import (
+	"context"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethersphere/bee/pkg/log"
+	"github.com/ethersphere/bee/pkg/settlement/swap/erc20"
+	"github.com/ethersphere/bee/pkg/storage"
+	"github.com/ethersphere/bee/pkg/storageincentives/redistribution"
+	"github.com/ethersphere/bee/pkg/swarm"
+	"math/big"
+	"sync"
+)
+
+const redistributionStatusKey = "redistribution_state"
+const loggerNameNode = "nodestatus"
+
+type NodeState struct {
+	stateStore     storage.StateStorer
+	erc20Service   erc20.Service
+	logger         log.Logger
+	overlay        swarm.Address
+	mtx            sync.Mutex
+	status         Status
+	currentBalance *big.Int
+	contract       redistribution.Contract
+}
+
+// Status provide internal status of the nodes in the redistribution game
+type Status struct {
+	Phase        PhaseType `json:"phase"`
+	State        State     `json:"state"`
+	Round        uint64    `json:"round"`
+	LastWonRound uint64    `json:"lastWonRound"`
+	Block        uint64    `json:"block"`
+	Reward       *big.Int  `json:"reward"`
+	Fees         *big.Int  `json:"fees"`
+}
+
+type State int
+
+const (
+	winner State = iota + 1
+	frozen
+	idle
+)
+
+func (s State) String() string {
+	switch s {
+	case winner:
```

Comment:
we can now remove the winner state as we have the lastWonRound

Reply:
Done